oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java

```diff
@@ -1470,6 +1470,7 @@ private UnionQueryImpl newAlternativeUnionQuery(@NotNull Query left, @NotNull Qu
         u.setMeasure(measure);
         u.setInternal(isInternal);
         u.setQueryOptions(queryOptions);
+        u.setOrderings(this.getOrderings());
```

Comment:
This is fine, or (for consistency): 

    u.setOrderings(orderings);

Reply:
Done